synapse/storage/databases/main/state.py

```diff
@@ -82,14 +79,17 @@ def get_room_version_txn(
                 removed from Synapse.
         """
         room_version_id = self.get_room_version_id_txn(txn, room_id)
-        v = KNOWN_ROOM_VERSIONS.get(room_version_id)
+        return self._retrieve_and_check_room_version(room_id, room_version_id)
 
+    def _retrieve_and_check_room_version(
```

Comment:
make this a top-level function rather than a member method? (it's fractionally faster to call a top-level function than a method)

Reply:
Hmm..., don't you think, it decreases code readability?

Reply:
I think it would be fine to make this a top-level function here; arguably it might *improve* readability since this method takes a `self` reference that it doesn't use (but from the outside it's not always clear why it does!).
It's also only 60 lines into the code unit, so it wouldn't be that far away anyway